### PR TITLE
Avoid exponential quoting in repr of deeply nested objects

### PR DESCRIPTION
**Pull request by encukou**
The `result` includes quoted strings inside other strings.
Each layer's quotes were backslash-escaped, each nesting layer
doubling the number of backslashes.

Simplify `result`, since it's meant for humans:
- Include the value literally, so it doesn't need quoting
- Omit the repr entirely if it's over 500 characters long

An artificial test case (pure_marshal/nested_tuples) and a real one
(test_parser.cpython-36) is added.

Fixes: https://github.com/fedora-python/marshalparser/issues/12

### CODE REVIEW

File: marshalparser/marshalparser.py [hunk 1]
```diff
@@ -189,7 +189,10 @@ def read_object(self) -> Any:
             index = self.read_long()
             self.references.append(Reference(byte=i, index=index))
             self.flag_refs[index].usages += 1
-            result = f"REF to {index}: " + str(self.flag_refs[index])
+            if len(str(self.flag_refs[index])) > 500:
+                result = ("REF to {index} (value omitted)")
```
Comment (hroncok) [inline marshalparser/marshalparser.py:193]: In this case, this is a string and the `()` is moot. Is that intentional?

Comment (frenzymadness) [reply marshalparser/marshalparser.py]: Black removed that in the latest commit.

**Comment by frenzymadness** [comment]
Thank you! The only problem in CI was formating. Fixed in the latest commit. I'll merge it and release a new version soon.

**Review by frenzymadness — APPROVED** [review]

**Comment by encukou** [comment]
Thanks!
`tox -e black` goes on and on complaining about everything in my venv :(  I guess you have set this up better.

**Pull request merged**